pennylane/operation.py

```diff
@@ -3017,6 +3018,22 @@ def convert_to_opmath(op):
     return op
 
 
+@contextmanager
+def use_new_opmath():
+    r"""Allows to use the new arithmetic operator dunders within a
+    temporary context using the `with` statement."""
+
+    check_op_math = active_new_opmath()
+    try:
+        enable_new_opmath()
+        yield
+    except Exception as e:
+        raise e
+    finally:
+        if not check_op_math:
+            disable_new_opmath()
+
```

Comment:
Something we can align with @Qottmann and @mudit2812 on. We do plan for op arithmetic to be enabled by default in 0.36, so this context manager could still serve a purpose in 0.35.

Reply:
Is there an advantage to use this over just
```python
active_new_opmath()
# code
disable_new_optmath()
```
or just extra style points 😎 ?

Reply:
If the user had enabled new operator arithmetic for their environment, then simply calling `disable_new_opmath()` might cause a side effect and turn it off for them. One way to deal with this would be to check if it was initially active before calling the disabling method.

``` py
active_flag = active_new_opmath()
enable_new_opmath()
# code
if not active_flag:
    disable_new_opmath()
```

A cleaner (and pythonic) way of doing this would be to use context manager so that the toggling of operator arithmetic happens within that context block (defined via `with`) and prevents any side effect beyond it -

``` py
with use_new_opmath():
    #code
```